### PR TITLE
Feat(eos_designs): Add metadata for Zscaler internet-exit

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Add metadata for Zscaler internet-exit

## Component(s) name

`arista.avd.eos_designs`
`arista.avd.cv_workflow`

## How to test

- Metadata is added in structured config.
- Not possible to test to CV until the CV side is done.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/plugins/plugin_utils/cv_client/workflows/deploy_cv_pathfinder_metadata_to_cv.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 from copy import deepcopy
 from logging import getLogger
 
+from ...password_utils.password import simple_7_decrypt
```
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/plugin_utils/cv_client/workflows/deploy_cv_pathfinder_metadata_to_cv.py:9]: do we need to decrypt? Can't this be down on CV side?

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/metadata/cv_pathfinder.py [hunk 2]
```diff
@@ -199,3 +199,12 @@ def _get_vni_for_vrf_name(self: AvdStructuredConfigMetadata, vrf_name: str):
             raise AristaAvdError(f"Unable to find the WAN VNI for VRF {vrf_name} during generation of cv_pathfinder metadata.")
 
         return wan_vni
+
+    def _metadata_internet_exit_policies(self: AvdStructuredConfigMetadata):
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/metadata/cv_pathfinder.py:203]: This changes the order of inserting the metadata in structured_config - not necessarily bad but still to consider?

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/metadata.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                    "country": get(
+                        self.shared_utils.wan_site,
+                        "resolved_location.country",
+                        required=True,
+                        org_key="cv_pathfinder_regions[].sites[].resolved_location.country",
+                    ),
+                    "upload_bandwidth": get(internet_exit_policy, "zscaler.upload_bandwidth"),
+                    "download_bandwidth": get(internet_exit_policy, "zscaler.download_bandwidth"),
+                    "firewall": get(internet_exit_policy, "zscaler.firewall.enabled", default=False),
+                    "ips_control": get(internet_exit_policy, "zscaler.firewall.ips", default=False),
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/metadata.py:54]: We have defaults here but not in the schema

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/metadata.schema.yml [hunk 4]
```diff
@@ -180,3 +180,46 @@ keys:
...
+                city:
+                  type: str
+                country:
+                  type: str
+                upload_bandwidth:
+                  type: int
+                download_bandwidth:
+                  type: int
+                firewall:
+                  type: bool
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/metadata.schema.yml:207]: default in the code (False)

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py [hunk 5]
```diff
@@ -719,3 +720,25 @@ def get_internet_exit_connections(self, internet_exit_policy: dict) -> list:
...
+        ipsec_key = self._generate_ipsec_key(name=policy_name, salt=ipsec_key_salt)
+        ufqdn = f"{self.shared_utils.hostname}_{policy_name}@{domain_name}"
+        return ufqdn, ipsec_key
+
+    def _generate_ipsec_key(self, name: str, salt: str) -> str:
+        """
+        Build a secret containing various components for this policy and device.
+        Run type-7 obfuscation using a algorithmic salt so we ensure the same key every time.
+
+        TODO: Maybe introduce some formatting with max length of each element, since the keys can be come very very long.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py:740]: the limit in EOS is quite crazy..  we should maybe use a sane one

File: ansible_collections/arista/avd/plugins/plugin_utils/cv_client/workflows/deploy_cv_pathfinder_metadata_to_cv.py (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/cv_client/workflows/deploy_cv_pathfinder_metadata_to_cv.py:9]: `Can't this be down on CV side?` No.
We will move to using CV "SecretService" in a later stage.

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/metadata/cv_pathfinder.py (hunk 2 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/metadata/cv_pathfinder.py:203]: Structured config is always ok to reorder.

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/metadata.py (hunk 3 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/metadata.py:54]: Right, I made the schema as lightweight as possible, but I can add some more details to it.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/metadata.schema.yml (hunk 4 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/metadata.schema.yml:207]: Not really. The schema describes the data model between eos_designs and cv_deploy/cv_workflow. The default values are implemented inside eos_designs. I have updated the schema with all the relevant values being required.

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/metadata.py (hunk 3 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/metadata.py:54]: added required fields. The defaults here are on the output, not input, so should not really be in the schema.

**Review by gmuloc — APPROVED** [review]

**Pull request merged**